src/main/kotlin/no/nav/syfo/brev/behandler/BehandlerVarselService.kt

```diff
@@ -43,54 +43,120 @@ class BehandlerVarselService(
-        when (varseltype) {
-            MotedeltakerVarselType.INNKALT -> opprettInnkallingVarselSvar(
+        val pMotedeltakerBehandlerVarsel = getBehandlerVarselForSvar(
+            varseltype = varseltype,
+            arbeidstakerPersonIdent = arbeidstakerPersonIdent,
+            conversationRef = conversationRef,
+            parentRef = parentRef,
+        )
+        pMotedeltakerBehandlerVarsel?.let {
+            database.createMotedeltakerBehandlerVarselSvar(
+                motedeltakerBehandlerVarselId = pMotedeltakerBehandlerVarsel.id,
                 type = svarType,
                 tekst = svarTekst,
-                conversationRef = conversationRef
             )
-            MotedeltakerVarselType.NYTT_TID_STED -> opprettEndringVarselSvar(
-                type = svarType,
-                tekst = svarTekst,
+            log.info("Created svar $svarType på varsel $varseltype with uuid ${pMotedeltakerBehandlerVarsel.uuid}")
+            return true
+        }
+
+        log.error("Could not find varsel of type $varseltype for conversationRef $conversationRef and parentRef $parentRef - Could not create svar")
+        return false
+    }
+
+    private fun getBehandlerVarselForSvar(
+        varseltype: MotedeltakerVarselType,
+        arbeidstakerPersonIdent: PersonIdentNumber,
+        conversationRef: UUID?,
+        parentRef: UUID?,
+    ): PMotedeltakerBehandlerVarsel? {
+        return when (varseltype) {
+            MotedeltakerVarselType.INNKALT -> getBehandlerVarselInnkalling(
+                arbeidstakerPersonIdent = arbeidstakerPersonIdent,
                 conversationRef = conversationRef,
-                parentRef = parentRef
             )
-            else -> throw IllegalArgumentException("Could not create svar for varsel $varseltype")
+            MotedeltakerVarselType.NYTT_TID_STED -> getBehandlerVarselNyttTidSted(
+                arbeidstakerPersonIdent = arbeidstakerPersonIdent,
+                conversationRef = conversationRef,
+                parentRef = parentRef,
+            )
+            else -> throw IllegalArgumentException("Cannot create svar for varsel $varseltype")
         }
     }
 
-    private fun opprettInnkallingVarselSvar(
-        type: DialogmoteSvarType,
-        tekst: String?,
-        conversationRef: UUID?
-    ) {
-        val pMotedeltakerBehandlerVarsel =
-            conversationRef?.let { database.getMotedeltakerBehandlerVarselForUuid(it) }
-        if (pMotedeltakerBehandlerVarsel?.varselType == MotedeltakerVarselType.INNKALT) {
-            database.createMotedeltakerBehandlerVarselSvar(
-                motedeltakerBehandlerVarselId = pMotedeltakerBehandlerVarsel.id,
-                type = type,
-                tekst = tekst,
+    private fun getBehandlerVarselInnkalling(
+        arbeidstakerPersonIdent: PersonIdentNumber,
+        conversationRef: UUID?,
+    ): PMotedeltakerBehandlerVarsel? {
+        val varselInnkallingForConversationRef =
```

Comment:
Litt usikker på om vi også burde sjekke at arbeidstakerPersonIdent stemmer når vi slår opp på conversationRef / parentRef.

Reply:
Ja, jeg synes vi bør sjekke at arbeidstakerPersonIdent også stemmer overens.

Reply:
Blir jo en del ekstra joins for å finne arbeidstakeren, men er kanskje tryggest ja 👍

Reply:
Fikset nå